### PR TITLE
[CBRD-24906]Add new testcase for The problem of performing a cross join by changing to the '0 <> 0' condition when the WHERE clause is always false

### PR DESCRIPTION
**Pull request by tw-kang**
http://jira.cubrid.org/browse/CBRD-24906

### CODE REVIEW

File: sql/_13_issues/_23_1h/cases/cbrd_24906.sql [hunk 1]
```diff
@@ -0,0 +1,341 @@
+drop table if exists tbl_a, tbl_b, tbl_c;
+create table tbl_a (col_a int primary key, col_b int, col_c int);
+create table tbl_b (col_a int primary key, col_b int, col_c int);
+create table tbl_c (col_a int primary key, col_b int, col_c int);
+insert into tbl_a select rownum, rownum, rownum from db_class a, db_class b, db_class c limit 1000;
+insert into tbl_b select rownum, rownum, rownum from db_class a, db_class b, db_class c limit 1000;
+insert into tbl_c select rownum, rownum, rownum from db_class a, db_class b, db_class c limit 1000;
+
```
Comment (youngjinj) [inline sql/_13_issues/_23_1h/cases/cbrd_24906.sql]: I think it is necessary to update statistical information.

File: sql/_13_issues/_23_1h/cases/cbrd_24906.sql [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+  (
+    select /*+ no_merge */
+      a.*
+    from
+      tbl_a as a, tbl_b as b
+    where a.col_a = b.col_a
+      and a.col_c = 1
+      and a.col_c = 9
+  );
+show trace;
```
Comment (youngjinj) [inline sql/_13_issues/_23_1h/cases/cbrd_24906.sql:25]: "I think that checking only the execution plan is sufficient for this test."

File: sql/_13_issues/_23_1h/cases/cbrd_24906.sql (hunk 1 above)
Comment (tw-kang) [reply sql/_13_issues/_23_1h/cases/cbrd_24906.sql]: review reflected.

File: sql/_13_issues/_23_1h/cases/cbrd_24906_1.sql [hunk 3]
```diff
@@ -0,0 +1,345 @@
...
+
+set trace on;
+
+
+-- test case 1
+select /*+ recompile */
+  count (*)
+from
+  (
+    select /*+ no_merge */
```
Comment (youngjinj) [inline sql/_13_issues/_23_1h/cases/cbrd_24906_1.sql]:
```suggestion
    select /*+ no_merge ordered */
```

File: sql/_13_issues/_23_1h/cases/cbrd_24906_1.sql [hunk 4]
```diff
@@ -0,0 +1,345 @@
...
+  );
+show trace;
+
+
+-- test case 2
+select /*+ recompile */
+  t.col_a
+from
+  (
+    select /*+ no_merge */
```
Comment (youngjinj) [inline sql/_13_issues/_23_1h/cases/cbrd_24906_1.sql]:
```suggestion
    select /*+ no_merge ordered */
```

File: sql/_13_issues/_23_1h/cases/cbrd_24906_1.sql [hunk 5]
```diff
@@ -0,0 +1,345 @@
...
+limit 999, 1;
+show trace;
+
+
+-- test case 3
+select /*+ recompile */
+  count (*)
+from
+  (
+    select /*+ no_merge */
```
Comment (youngjinj) [inline sql/_13_issues/_23_1h/cases/cbrd_24906_1.sql]:
```suggestion
    select /*+ no_merge ordered */
```

File: sql/_13_issues/_23_1h/cases/cbrd_24906_1.sql [hunk 6]
```diff
@@ -0,0 +1,345 @@
...
+  );
+show trace;
+
+
+-- test case 4
+select /*+ recompile */
+  count (*)
+from
+  (
+    select /*+ no_merge */
```
Comment (youngjinj) [inline sql/_13_issues/_23_1h/cases/cbrd_24906_1.sql]:
```suggestion
    select /*+ no_merge ordered */
```

File: sql/_13_issues/_23_1h/cases/cbrd_24906_1.sql [hunk 7]
```diff
@@ -0,0 +1,345 @@
...
+  );
+show trace;
+
+
+-- test case 5
+select /*+ recompile */
+  count(*)
+from
+  (
+    select /*+ no_merge */
```
Comment (youngjinj) [inline sql/_13_issues/_23_1h/cases/cbrd_24906_1.sql]:
```suggestion
    select /*+ no_merge ordered */
```

File: sql/_13_issues/_23_1h/cases/cbrd_24906_1.sql [hunk 8]
```diff
@@ -0,0 +1,345 @@
...
+show trace;
+
+
+-- test case 9
+select /*+ recompile */
+  count (*)
+from
+  tbl_a as a
+where
+  exists (select /*+ no_merge */ 1 from tbl_b as b, tbl_c as c where a.col_a = b.col_a and b.col_a = c.col_a and c.col_c = 1 and c.col_c = 9);
```
Comment (youngjinj) [inline sql/_13_issues/_23_1h/cases/cbrd_24906_1.sql]:
```suggestion
  exists (select /*+ no_merge ordered */ 1 from tbl_b as b, tbl_c as c where a.col_a = b.col_a and b.col_a = c.col_a and c.col_c = 1 and c.col_c = 9);
```

File: sql/_13_issues/_23_1h/cases/cbrd_24906_1.sql [hunk 9]
```diff
@@ -0,0 +1,345 @@
...
+show trace;
+
+
+-- test case 12 
+select /*+ recompile */
+  count (*)
+from
+  tbl_a as a
+where
+  a.col_a in (select /*+ no_merge */ c.col_a from tbl_b as b, tbl_c as c where b.col_a = c.col_a and c.col_c = 1 and c.col_c = 9);
```
Comment (youngjinj) [inline sql/_13_issues/_23_1h/cases/cbrd_24906_1.sql]:
```suggestion
  a.col_a in (select /*+ no_merge ordered */ c.col_a from tbl_b as b, tbl_c as c where b.col_a = c.col_a and c.col_c = 1 and c.col_c = 9);
```

File: sql/_13_issues/_23_1h/cases/cbrd_24906_1.sql [hunk 10]
```diff
@@ -0,0 +1,345 @@
...
+  );
+show trace;
+
+
+-- test case 19 
+select /*+ recompile */
+  count (*)
+from
+  (
+    select /*+ no_merge */
```
Comment (youngjinj) [inline sql/_13_issues/_23_1h/cases/cbrd_24906_1.sql]:
```suggestion
    select /*+ no_merge ordered */
```

File: sql/_13_issues/_23_1h/cases/cbrd_24906_1.sql [hunk 11]
```diff
@@ -0,0 +1,345 @@
...
+  ) as t;
+show trace;
+
+
+-- test case 20 
+select /*+ recompile */
+  count (*)
+from
+  (
+    select /*+ no_merge */
```
Comment (youngjinj) [inline sql/_13_issues/_23_1h/cases/cbrd_24906_1.sql]:
```suggestion
    select /*+ no_merge ordered */
```

File: sql/_13_issues/_23_1h/cases/cbrd_24906_1.sql [hunk 12]
```diff
@@ -0,0 +1,345 @@
...
+  ) as t;
+show trace;
+
+
+-- test case 21 
+select /*+ recompile */
+  count (*)
+from
+  (
+    select /*+ no_merge */
```
Comment (youngjinj) [inline sql/_13_issues/_23_1h/cases/cbrd_24906_1.sql]:
```suggestion
    select /*+ no_merge ordered */
```

File: sql/_13_issues/_23_1h/cases/cbrd_24906_1.sql [hunk 13]
```diff
@@ -0,0 +1,345 @@
...
+  ) as t;
+show trace;
+
+
+-- test case 22 
+select /*+ recompile */
+  count (*)
+from
+  (
+    select /*+ no_merge */
```
Comment (youngjinj) [inline sql/_13_issues/_23_1h/cases/cbrd_24906_1.sql]:
```suggestion
    select /*+ no_merge ordered */
```

File: sql/_13_issues/_23_1h/cases/cbrd_24906_1.sql [hunk 14]
```diff
@@ -0,0 +1,345 @@
...
+  ) as t;
+show trace;
+
+
+-- test case 23 
+select /*+ recompile */
+  count (*)
+from
+  (
+    select /*+ no_merge */
```
Comment (youngjinj) [inline sql/_13_issues/_23_1h/cases/cbrd_24906_1.sql]:
```suggestion
    select /*+ no_merge ordered */
```

File: sql/_13_issues/_23_1h/cases/cbrd_24906_1.sql [hunk 15]
```diff
@@ -0,0 +1,345 @@
...
+show trace;
+
+
+-- test case 24 
+prepare q from '
+select /*+ recompile */
+  count(*)
+from
+  (
+    select /*+ no_merge */
```
Comment (youngjinj) [inline sql/_13_issues/_23_1h/cases/cbrd_24906_1.sql]:
```suggestion
    select /*+ no_merge ordered */
```

**Comment by youngjinj** [comment]
`/*+ ORDERED */` hint needs to be added for consistency in the join order in the answer.
Changes in the answer are expected.

File: sql/_13_issues/_23_1h/cases/cbrd_24906_1.sql (hunk 3 above)
Comment (tw-kang) [reply sql/_13_issues/_23_1h/cases/cbrd_24906_1.sql]: The review has been reflected.

**Review by youngjinj — APPROVED** [review]

**Review by ssihil — APPROVED** [review]

**Review by hiclass — APPROVED** [review]

**Review by kwonhoil — APPROVED** [review]

**Pull request merged**